### PR TITLE
Add support for formatting AddrOf (unary "&").

### PR DESCRIPTION
**Pull request by eefriedman**

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -1414,3 +1413,16 @@ pub fn rewrite_assign_rhs<S: Into<String>>(context: &RewriteContext,
...
     Some(result)
 }
+
+fn rewrite_expr_addrof(context: &RewriteContext,
+                       mutability: ast::Mutability,
+                       expr: &ast::Expr,
+                       width: usize,
+                       offset: Indent)
+                       -> Option<String> {
+    let operator_str: &'static str = match mutability {
```
Comment (nrc) [inline src/expr.rs]: do we need the type here?

File: tests/source/expr.rs [hunk 2]
```diff
@@ -298,3 +298,7 @@ fn returns() {
 
     return aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa + aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
 }
+
+fn addrof() {
+    &mut (aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb)
```
Comment (nrc) [inline tests/source/expr.rs]: may as well get the whitespace around `mut` wrong in the source.

Also could you add an example without `mut`.

**Comment by nrc** [comment]
r+ with the minor comments addressed

**Comment by eefriedman** [comment]
Updated to address review comments.

**Comment by nrc** [comment]
Great, thank you!

**Pull request merged**